src/libp2p_transport_tcp.erl

```diff
@@ -471,7 +471,8 @@ handle_info({stungun_nat, TxnID, NatType}, State=#state{tid=TID, stun_txns=StunT
             lager:debug("stungun detected NAT type ~p", [NatType]),
             case NatType of
                 none ->
-                    %% don't need to start relay here
+                    %% don't need to start relay here, stop any we already have
+                    libp2p_relay_server:stop(libp2p_swarm:swarm(TID)),
```

Comment:
if this server was started then stopped should we remove relay address from peerbook?

Reply:
The terminate for the relay server handles that

Reply:
yep I missed the `true = libp2p_config:remove_relay(TID),`